hubconf.py

```diff
@@ -0,0 +1,128 @@
+"""
+Configuration file for loading pretrained models using PyTorch hub
+Usage example: torch.hub.load("castorini/howl", "hey_fire_fox")
+"""
+dependencies = ['howl', 'torch']
+
+import dataclasses
+import os
+import pathlib
+import shutil
+import typing
+import zipfile
+
+import torch
+
+import howl.context as context
+import howl.data.transform as transform
+import howl.model as howl_model
+import howl.model.inference as inference
+import howl.settings as settings
+
+from howl.settings import SETTINGS as _SETTINGS
+
+_MODEL_URL = "https://github.com/castorini/howl-models/archive/v1.0.0.zip"
+_MODEL_CACHE_FOLDER = "howl-models"
+
+@dataclasses.dataclass
+class _ModelSettings:
+    path: str # Relative to the howl-models root folder
+    model_name: str
+    vocab: typing.List[str]
+    inference_sequence: typing.List[int]
+    token_type: str
+    use_frame: bool
+
+
+def hey_fire_fox(pretrained=True, **kwargs) \
+        -> typing.Tuple[inference.InferenceEngine, context.InferenceContext]:
+    """
+    Pretrained model for Firefox Voice
+    """
+    # Audio inference settings
+    conf = _ModelSettings(path="howl/hey-fire-fox",
+                          model_name="res8",
+                          vocab=["hey", "fire", "fox"],
+                          inference_sequence=[0, 1, 2],
+                          token_type="word",
+                          use_frame=True)
+    ats = transform.augment.AudioTransformSettings(num_mels=40)
+    _SETTINGS._training = settings.TrainingSettings(max_window_size_seconds=0.5)
+
+    # Set cache directory and download models
+    cache_dir = pathlib.Path.home() / '.cache/howl'
+    if not os.path.exists(cache_dir):
+        os.makedirs(cache_dir)
+
+    # Separate `reload_models` flag since PyTorch will pop the 'force_reload' flag
+    reload_models = kwargs.pop('reload_models', False)
+    _download_howl_models(cache_dir, reload_models)
+
+    # Set up context and workspace
+    ws_path: pathlib.Path = cache_dir / _MODEL_CACHE_FOLDER / conf.path
+    ctx = context.InferenceContext(
+        conf.vocab, token_type=conf.token_type, use_blank=not conf.use_frame)
+    ws = howl_model.Workspace(ws_path, delete_existing=False)
```

Comment:
check this object for getting the right parameters (num_mels, inference sequence etc)